geode-management/src/main/java/org/apache/geode/management/api/ClusterManagementResult.java

```diff
@@ -131,15 +148,35 @@ public String getStatusMessage() {
    * Returns the full path (not including http://server:port) by which this result can be referenced
    * via REST
    */
+  @JsonIgnore
   public String getUri() {
-    return uri;
+    return (links == null) ? null : links.get(Links.SELF);
+  }
+
+  /**
+   * for internal use only
```

Comment:
I prefer we would not have so many "for internal use only" methods. I can understand we would have a few for json serialization or  convenience method. For this story, do we need to have a top level links?

Reply:
these seem like 2 unrelated issues.  public internal methods can be solved by changing our package structure or by moving to Jigsaw modules, otherwise I don't see this as any worse than countless other places.  If you want this split into multiple PRs, that's fine but should have nothing to do with internal method scoping...

Reply:
PR was split so this is no longer relevant